### PR TITLE
Detect in-progress file transfers before processing

### LINKED ISSUE #36: File watcher: detect in-progress file transfers before processing
## Context

The `DirectoryWatchingService` processes files as soon as a filesystem event fires. When a large video file is being copied or downloaded into a watched directory, the watcher may attempt to process the file before the transfer is complete, leading to incomplete or corrupt media file records.

## Requirements

- Detect when a file is still being written to (growing in size)
- Wait for the file to stabilize (e.g., no size change for N seconds) before processing
- Handle common scenarios: network copies, torrent downloads, NFS transfers
- Configurable stabilization period (default: 10-30 seconds)

## Notes

Previously tracked as a TODO in `DirectoryWatchingService`.

### PR DESCRIPTION
**Pull request by Stuckya**
## Summary

Closes #36

- Add `FileStabilityChecker` that polls file size until stable before allowing processing, preventing corrupt media file records from incomplete transfers
- Wire stability checking into `DirectoryWatchingService` with extension validation, deduplication via `ConcurrentHashMap.computeIfAbsent`, library resolution by longest path prefix match, and virtual thread execution
- Add `processDiscoveredFile` public API to `LibraryManagementService` for watcher-driven file processing
- Add externalized `LibraryWatcherProperties` configuration (stabilization period, poll interval, max wait)
- Fix `resolveLibrary` bug where `String.startsWith` falsely matched paths sharing a string prefix but not a path prefix (e.g., `/media/movies` matching `/media/moviesfoo`)

## Test plan

- [x] 33 new unit tests across 4 test files covering all edge cases (file stable/unstable, missing, deleted mid-check, max wait exceeded, zero-byte, interrupted, deduplication, library resolution, exception cleanup, OVERFLOW)
- [x] `./mvnw test` — 518 total tests, 0 new failures
- [x] `./mvnw spotless:check` — formatting passes
- [ ] Manual: start app with a library configured, copy a large file into the library directory, observe logs showing stability polling and eventual processing after stabilization

### CODE REVIEW

File: src/main/java/com/streamarr/server/services/library/FileEventProcessor.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    if (task != null) {
+      task.future().cancel(true);
+      log.info("Cancelled in-flight check for deleted file: {}", path);
+    }
+    log.info("Watcher event type: DELETE -- filepath: {}", path);
+  }
+
+  private Optional<UUID> resolveLibrary(Path path) {
+    var absolutePath = path.toAbsolutePath();
+    var fs = absolutePath.getFileSystem();
```
[the remark this answers is not in the document]
Comment (Stuckya) [reply src/main/java/com/streamarr/server/services/library/FileEventProcessor.java]: Per CLAUDE.md defensive programming guidelines: *"Validate inputs at the API boundary; trust internal code."*

The `path` argument originates from the file watcher (system boundary already passed). Calling `toAbsolutePath()` on a valid `Path` always returns a valid `Path` with an associated `FileSystem`. Adding null/validation checks here would be over-defensive for internal code.

No change needed.
